superset/connectors/sqla/models.py

```diff
@@ -373,7 +373,11 @@ class SqlaTable(Model, BaseDatasource):
     owner_class = security_manager.user_model
 
     __tablename__ = "tables"
-    __table_args__ = (UniqueConstraint("database_id", "table_name"),)
+    __table_args__ = (
+        UniqueConstraint(
+            "database_id", "schema", "table_name", name="_customer_location_uc",
+        ),
+    )
```

Comment:
Ping @john-bodley . I've been meaning to bring this up for discussion, as I also feel schema should be included in the constraint. However, I remember you had done work in this area before, and there was some reason why it wasn't possible to include schema in the unique constraint.

Reply:
I don't think this is a viable solution based on how different engine treat uniqueness for nullable columns (`schema` in this case). This is documented in https://github.com/apache/incubator-superset/pull/5449 which is an open PR from over two years ago but was updated last year to remedy the nullable issue. Though it's not perfect, i.e., ideally the constraint should reside within the database model I sense it's the only viable solution unless we special case the various engines.

Reply:
hey @john-bodley @villebro , sorry to bring this back, but how about we just get rid of the constraint altogether? You want 2 datasets pointing to the same table, why not?

At least we should alter the old migration that creates the old/bad constraint we then cannot delete successfuly

Reply:
Having multiple Superset datasources with the same schema and table name for a given database does not sense to me as it is not easily apparent from the current UI which datasource is being referenced as the name if the primary indicator throughout the application.